dlc/src/lib.rs

```diff
@@ -0,0 +1,1029 @@
+    /// If input amount (sum of all input values) is lower than the sum of the collateral
+    /// plus the required fees, an error is returned.
+    pub fn get_change_output_and_fees(
+        &self,
+        fee_rate_per_vb: u64,
+    ) -> Result<(TxOut, u64, u64), Error> {
+        let inputs_weight: usize = self
+            .inputs
+            .iter()
+            .map(|i| {
+                let script_size = util::redeem_script_to_script_sig(&i.redeem_script).len();
+                // Base tx input weight + redeem script weight (*4 because non witness)
+                // + max witness weight (*1 because witness)
+                TX_INPUT_BASE_WEIGHT + 4 * script_size + i.max_witness_len
+            })
+            .sum();
+
+        // Value size + script length var_int + ouput script pubkey size
+        let change_size = self.change_script_pubkey.len();
+        // Change size is scaled by 4 from vBytes to weight units
+        let change_weight = change_size * 4;
+
+        // Base weight (nLocktime, nVersion, ...) is distributed among parties
+        // independently of inputs contributed
+        let this_party_fund_base_weight = FUND_TX_BASE_WEIGHT / 2;
+
+        let total_fund_weight = this_party_fund_base_weight + inputs_weight + change_weight + 36;
+        let fund_fee = util::weight_to_fee(total_fund_weight, fee_rate_per_vb);
+
+        // Base weight (nLocktime, nVersion, funding input ...) is distributed
+        // among parties independently of output types
+        let this_party_cet_base_weight = CET_BASE_WEIGHT / 2;
+
+        // size of the final script pubkey scaled by 4 from vBytes to weight units
+        let output_spk_weight = self.final_script_pubkey.len() * 4;
+        let total_cet_weight = this_party_cet_base_weight + output_spk_weight;
+        let cet_or_refund_fee = util::weight_to_fee(total_cet_weight, fee_rate_per_vb);
+        let required_input_funds = self.collateral + fund_fee + cet_or_refund_fee;
+        if self.input_amount < required_input_funds {
+            return Err(Error::InvalidArgument);
+        }
+
+        let change_output = TxOut {
+            value: self.input_amount - required_input_funds,
+            script_pubkey: self.change_script_pubkey.clone(),
+        };
+
+        Ok((change_output, fund_fee, cet_or_refund_fee))
+    }
+
+    fn get_unsigned_tx_inputs(&self, sequence: u32) -> Vec<TxIn> {
+        self.inputs
+            .iter()
+            .map(|i| TxIn {
+                previous_output: i.outpoint,
+                script_sig: util::redeem_script_to_script_sig(&i.redeem_script),
+                sequence,
+                witness: Vec::new(),
+            })
+            .collect()
+    }
+}
+
+/// Create the transactions for a DLC contract based on the provided parameters
+pub fn create_dlc_transactions(
+    offer_params: &PartyParams,
+    accept_params: &PartyParams,
+    payouts: &[Payout],
+    refund_lock_time: u32,
+    fee_rate_per_kw: u64,
```

Comment:
I think this also has to be `fee_rate_per_vb`

Reply:
Thanks indeed!